frontend/src/plugin/registry.tsx

```diff
@@ -124,44 +134,272 @@ export default class Registry {
     store.dispatch(setDetailsView(sectionName, sectionFunc));
   }
+
   /**
-   * @param component is a React Component that takes two required props ```JSX logoType``` which is a
-   * constant string literal that accepts either of the two values ```JSX small``` or ```JSX large``` depending on whether
-   * the sidebar is in shrink or expaned state so that you can change your logo from small to large and the other optional
-   * prop is the ```JSX themeName``` which is a string with two values 'light' and 'dark' base on which theme is selected.
+   * Add a logo for Headlamp to use instead of the default one.
    *
+   * @param logo is a React Component that takes two required props
+   * `logoType` which is a constant string literal that accepts either
+   * of the two values `small` or `large` depending on whether
+   * the sidebar is in shrink or expanded state so that you can change your logo
+   * from small to large and the other optional prop is the `themeName`
+   * which is a string with two values 'light' and 'dark' base on which theme is selected.
    *
    * @example
-   * ```JSX
-   * register.registerAppLogo((props: { logoType: 'small' | 'large', themeName: string}) => <MY_CUSTOM_COMPONENT logoType={logoType}/>)
+   * ```tsx
+   * import { registerAppLogo } from '@kinvolk/headlamp-plugin/lib';
+   * registerAppLogo(<p>my logo</p>)
    * ```
+   *
+   * More complete logo example in plugins/examples/change-logo:
+   * @see {@link http://github.com/kinvolk/headlamp/plugins/examples/change-logo/ Change Logo Example}
+   *
    */
-  registerAppLogo(component: BrandingProps['logo']) {
-    store.dispatch(setBrandingAppLogoComponent(component));
+  registerAppLogo(logo: AppLogoType) {
+    store.dispatch(setBrandingAppLogoComponent(logo));
   }
 
   /**
-   * @param component is a React Component that takes one required props ```JSX clickHandler``` which is the
+   * Use a custom cluster chooser button
+   *
+   * @param component is a React Component that takes one required props ```tsx clickHandler``` which is the
    * action handler that happens when the custom chooser button component click event occurs
+   *
    * @example
-   * ```JSX
-   * registry.registerClusterChooserComponent((props: { clickHandler: () => {} }) => <MY_CUSTOM_COMPONENT onClick={clickHandler}/>)
+   * ```tsx
+   * import { ClusterChooserProps, registerClusterChooser } from '@kinvolk/headlamp-plugin/lib';
+   *
+   * function MyClusterChooser(props: ClusterChooserProps) {
+   *   return <button onClick={clickHandler}>my chooser</button>;
+   * }
+   *
+   * registerClusterChooser(MyClusterChooser)
    * ```
+   *
+   * @see {@link http://github.com/kinvolk/headlamp/plugins/examples/clusterchooser/ Cluster Chooser Example}
+   *
    */
-  registerClusterChooserComponent(component: clusterChooserButtonComponent | null) {
+  registerClusterChooser(component: React.ComponentType<ClusterChooserProps> | null) {
```

Comment:
IIRC we had a way to override the cluster chooser (dialog) or the cluster chooser button.
Is this still possible?